drivers/include/pn532.h

```diff
@@ -0,0 +1,373 @@
+ * @param[in]  mode         new mode for the SAM
+ * @param[in]  timeout      timeout for Virtual Card mode with LSB of 50ms.
+ *                          (0 = infinite and 0xFF = 12.75s)
+ *
+ * @return                  0 on success
+ */
+int pn532_sam_configuration(pn532_t *dev, pn532_sam_conf_mode_t mode, unsigned timeout);
+
+/**
+ * @brief Get (max) passive targets of the given type
+ *
+ * This method blocks until a target is detected.
+ *
+ * @param[in]  dev          target device
+ * @param[in]  target       type of target
+ * @param[in]  max          maximum number of targets to detect (1 or 2)
+ * @param[in]  max_retries  maximum number of attempts to activate a target
+ *                          (0xff blocks indefinitely)
+ *
+ * @return                  0 on success
+ */
+int pn532_list_passive_targets(pn532_t *dev, pn532_target_t target, unsigned max,
+                               unsigned max_retries);
+
+/**
+ * @brief Get one ISO14443-A passive target
+ *
+ * This method blocks until a target is detected.
+ *
+ * @param[in]  dev          target device
+ * @param[out] out          target to be stored
+ * @param[in]  max_retries  maximum number of attempts to activate a target
+ *                          (0xff blocks indefinitely)
+ *
+ * @return                  0 on success
+ */
+int pn532_get_passive_iso14443a(pn532_t *dev, nfc_iso14443a_t *out, unsigned max_retries);
+
+/**
+ * @brief Authenticate a Mifare classic card
+ *
+ * This operation must be done before reading or writing the segment.
+ *
+ * @param[in]  dev          target device
+ * @param[in]  card         card to use
+ * @param[in]  keyid        which key to use
+ * @param[in]  key          buffer containing the key
+ * @param[in]  block        which block to authenticate
+ *
+ * @return                  0 on success
+ */
+int pn532_mifareclassic_authenticate(pn532_t *dev, nfc_iso14443a_t *card,
+                                     pn532_mifare_key_t keyid, char *key, unsigned block);
+
+/**
+ * @brief Read a block of a Mifare classic card
+ *
+ * The block size is 16 bytes and it must be authenticated before read.
+ *
+ * @param[in]  dev          target device
+ * @param[out] odata        buffer where to store the data
+ * @param[in]  card         card to use
+ * @param[in]  block        which block to read
+ *
+ * @return                  0 on success
+ */
+int pn532_mifareclassic_read(pn532_t *dev, char *odata, nfc_iso14443a_t *card, unsigned block);
+
+/**
+ * @brief Write a block of a Mifare classic card
+ *
+ * The block size is 16 bytes and it must be authenticated before written.
+ *
+ * @param[in]  dev          target device
+ * @param[in]  idata        buffer containing the data to write
+ * @param[in]  card         card to use
+ * @param[in]  block        which block to write to
+ *
+ * @return                  0 on success
+ */
+int pn532_mifareclassic_write(pn532_t *dev, char *idata, nfc_iso14443a_t *card, unsigned block);
```

Comment:
Never used but some people might be interested in it (way more interesting than the GPIOs I gues :-) )